core/src/types/eth_bridge_pool.rs

```diff
@@ -1,23 +1,117 @@
 //! The necessary type definitions for the contents of the
 //! Ethereum bridge pool
 
+use std::borrow::Cow;
+
 use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
 use ethabi::token::Token;
 use serde::{Deserialize, Serialize};
 
 use crate::ledger::eth_bridge::storage::wrapped_erc20s;
 use crate::types::address::Address;
 use crate::types::eth_abi::Encode;
-pub use crate::types::ethereum_events::TransferToEthereumKind;
 use crate::types::ethereum_events::{
     EthAddress, TransferToEthereum as TransferToEthereumEvent,
 };
+use crate::types::hash::Hash as HashDigest;
 use crate::types::storage::{DbKeySeg, Key};
 use crate::types::token::Amount;
 
+/// A version used in our Ethereuem smart contracts
+const VERSION: u8 = 1;
+
 /// A namespace used in our Ethereuem smart contracts
 const NAMESPACE: &str = "transfer";
 
+/// Transfer to Ethereum kinds.
+#[derive(
+    Copy,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    Hash,
+    PartialOrd,
+    Ord,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+    Serialize,
+    Deserialize,
+)]
+pub enum TransferToEthereumKind {
+    /// Transfer ERC20 assets from Namada to Ethereum.
+    ///
+    /// These transfers burn wrapped ERC20 assets in Namada, once
+    /// they have been confirmed.
+    Erc20,
+    /// Refund non-usable tokens.
+    ///
+    /// These Bridge pool transfers should be crafted for assets
+    /// that have been transferred to Namada, that had either not
+    /// been whitelisted or whose token caps had been exceeded in
+    /// Namada at the time of the transfer.
+    Nut,
+}
+
+/// Additional data appended to a [`TransferToEthereumEvent`] to
+/// construct a [`PendingTransfer`].
+#[derive(
+    Debug,
+    Clone,
+    Hash,
+    PartialOrd,
+    PartialEq,
+    Ord,
+    Eq,
+    Serialize,
+    Deserialize,
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+)]
+pub struct PendingTransferAppendix<'transfer> {
+    /// The kind of the pending transfer to Ethereum.
+    pub kind: Cow<'transfer, TransferToEthereumKind>,
+    /// The sender of the transfer.
+    pub sender: Cow<'transfer, Address>,
+    /// The amount of gas fees (in NAM)
+    /// paid by the user sending this transfer
+    pub gas_fee: Cow<'transfer, GasFee>,
+}
+
+impl From<PendingTransfer> for PendingTransferAppendix<'static> {
```

Comment:
I'm going to let this fancy pants micro-optimization stay, but this is a overkill for fields we could just clone.